Examples/Framework/include/ACTFW/Validation/DuplicationPlotTool.hpp

```diff
@@ -0,0 +1,90 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2019 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include <map>
+#include <memory>
+#include <string>
+
+#include "ACTFW/Utilities/Helpers.hpp"
+#include "Acts/Utilities/Logger.hpp"
+#include "ActsFatras/EventData/Particle.hpp"
+
+namespace FW {
+
+// Tools to make duplication number&rate plots to show tracking duplication.
+class DuplicationPlotTool {
+ public:
+  /// @brief The nested configuration struct
+  struct Config {
+    std::map<std::string, PlotHelpers::Binning> varBinning = {
+        {"Eta", PlotHelpers::Binning("#eta", 40, -4, 4)},
+        {"Phi", PlotHelpers::Binning("#phi", 100, -3.15, 3.15)},
+        {"Pt", PlotHelpers::Binning("pT [GeV/c]", 40, 0, 100)},
+        {"Num", PlotHelpers::Binning("N", 30, -0.5, 29.5)}};
+  };
+
+  /// @brief Nested Cache struct
+  struct DuplicationPlotCache {
+    TProfile* nDuplicated_vs_pT;         ///< Number of duplicated tracks vs pT
+    TProfile* nDuplicated_vs_eta;        ///< Number of duplicated tracks vs eta
+    TProfile* nDuplicated_vs_phi;        ///< Number of duplicated tracks vs phi
+    TEfficiency* duplicationRate_vs_pT;  ///< Tracking duplication rate vs pT
+    TEfficiency* duplicationRate_vs_eta;  ///< Tracking duplication rate vs eta
+    TEfficiency* duplicationRate_vs_phi;  ///< Tracking duplication rate vs phi
+  };
+
+  /// Constructor
+  ///
+  /// @param cfg Configuration struct
+  /// @param lvl Message level declaration
+  DuplicationPlotTool(const Config& cfg, Acts::Logging::Level lvl);
+
+  /// @brief book the duplication plots
+  ///
+  /// @param duplicationPlotCache the cache for duplication plots
+  void book(DuplicationPlotCache& duplicationPlotCache) const;
+
+  /// @brief fill duplication plots
+  ///
+  /// @param duplicationPlotCache cache object for duplication plots
+  /// @param truthParticle the truth Particle
+  /// @param status the reconstruction status
```

Comment:
So is this true if the particle is duplicated?

Reply:
Right. The reco (truth-matched) tracks could be classified into 'real' and 'duplicated'. If the track is duplicated (i.e. the matched truth particle is duplicated), the fill status will be true; otherwise false. So the duplication is in terms of each reco track. The `truthParticle` is used because the eta/pT information is needed for the plotting (I'm thinking of adding another method to use reco track parameters in #171).

Reply:
> I don't fully understand the two fill methods. This is used in #171 is that correct?

Do you mean the fill for fake rate in #171? It's different from the duplication rate. The fake rate is in terms of each reco (could be truth-unmatched) track while the duplication is in terms of each reco&truth-matched track.

Reply:
This `DuplicationPlotTool` has two fill methods if I see this correctly:

`void fill(DuplicationPlotCache& duplicationPlotCache, const ActsFatras::Particle& truthParticle, bool status) const;`

`void fill(DuplicationPlotCache& duplicationPlotCache, const ActsFatras::Particle& truthParticle,  size_t nDuplicatedTracks) const;` and I'm not sure I understand the difference is.

Reply:
Could you document the meaning of `status` then in the comment?

Reply:
Ah, I see. The first `fill` method fills the duplication rate (into `TEfficiency` object). The second `fill` fills the number of duplicated tracks (into a histogram object)

Reply:
The comments have been fixed.